arcaflow_plugin_horreum_client/horreum_client_schema.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import enum
+import typing
+import re
+from dataclasses import dataclass
+from arcaflow_plugin_sdk import schema
+
+
+class access(str, enum.Enum):
+    PUBLIC = "PUBLIC"
+    PROTECTED = "PROTECTED"
+    PRIVATE = "PRIVATE"
+
+
+@dataclass
+class InputParams:
+    horreum_url: typing.Annotated[
+        str,
+        schema.name("Horreum URL"),
+        schema.description(
+            "The complete base URL for the Horreum server,"
+            " such as 'https://horreum.example.com'."
+        ),
+        schema.pattern(
+            re.compile(
+                r"(https?:\/\/)?"
+                r"[a-zA-Z0-9_-]{1,63}(\.[a-zA-Z0-9_-]{1,63})*"
+                r"(:[0-9]{1,5})?"
+                r"\/?"
+            )
+        ),
+    ]
+    horreum_keycloak_url: typing.Annotated[
+        str,
+        schema.name("Horreum Keycloak URL"),
+        schema.description(
+            "The complete base URL for the Horreum Keycloak server,"
+            " such as 'https://keycloak.example.com'."
+        ),
+        schema.pattern(
+            re.compile(
+                r"(https?:\/\/)?"
+                r"[a-zA-Z0-9_-]{1,63}(\.[a-zA-Z0-9_-]{1,63})*"
+                r"(:[0-9]{1,5})?"
+                r"\/?"
+            )
+        ),
+    ]
+    horreum_username: typing.Annotated[
+        str,
+        schema.name("Horreum username"),
+        schema.description("Username for the Horreum server."),
+    ]
+    horreum_password: typing.Annotated[
+        str,
+        schema.name("Horreum password"),
+        schema.description("Password for the Horreum server."),
+    ]
```

Comment:
In "real world" you'd probably want a password stored in some sort of secure secret rather than inline in a YAML file. Does Horreum have the concept of persistent API keys that could be managed through the server and used for repeatable client operations like this? That'd be a much better way to do handle this. (I also wonder why "the" Horreum service inside Red Hat never bothered to hook up to Red Hat SSO, but that's a different issue. 😆 )

Reply:
We are discussing a proper secret vault solution for Arcaflow, but we don't have a real option as of today. And per above, Horreum isn't quite ready for the use of client keys. So I think we will live with this design in the near term until the updated Horreum design and the Arcaflow secret vault solutions are ready.